carbon-design-system/carbon
# feat: combobox's new feature, autocomplete with typeahead

Closes https://github.com/carbon-design-system/carbon/issues/14007

Added autocomplete and typeahead feature for combobox with custom filter to support the typeahead.

#### Changelog

**New**

- new `autocomplete` prop
- new `autocompleteCustomFunction` which is 1-to-1 mapping filter function which filters matching both character and position.
- new test to make sure `autocomplete` and `autocompleteCustomFunction` are working.


**Changed**

- added `getMenuProps` and `refs.setFloationg` to `useMemo` hook
- modified import for `act` in the test helpers

#### Testing / Reviewing

- navigate to new  "Autocomplete with Typeahead" story
- check that the suggestions are coming in the input box or not.
